UnityClient/Assets/Scenes/Map/MapUiController.cs

```diff
@@ -102,4 +107,27 @@ public class MapUiController : MonoBehaviour {
     public void HideTooltip() {
         Tooltip.SetText(null, Vector3.zero);
     }
+
+    public void OnRestartAnswer(ushort cmd, int size, InPacket packet) {
+        if (packet is ZC.RESTART_ACK) {
+            var pkt = packet as ZC.RESTART_ACK;
+            if (pkt.type == 0) {
+                ChatBox.DisplayMessage(502, ChatMessageType.ERROR);
+            }
+            else {
+                // @todo ?
+                // clear StatusIcons
+                // clear ChatBox
+                // clear ShortCut
+                // clear PartyFriends
+                // clear renderers
```

Comment:
This could be done if you put these things to observe your current Entity & Session. Once they are cleared, things clear themselves.

That is a pattern I've been wanting to implement on the project for a while now, it might be the right time to do it